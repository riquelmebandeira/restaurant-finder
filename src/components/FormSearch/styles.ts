import styled, { css } from "styled-components";

export const Form = styled.form`
  ${({theme}) => css`
    background-color: ${theme.colors.white};
    max-width: 54rem;
    height: 5rem;
    padding: 1.3rem 2rem;
    border: 1px solid ${theme.colors.lightMedium};
    border-radius: .8rem;
    display: flex;
    align-items: center;

    input {
      border: none;
      margin-left: 1rem;
      outline: none;
      width: 100%;

      &::placeholder {
        font-size: ${theme.font.sizes.paragraph};
        font-family: ${theme.font.family.default};
        color: ${theme.colors.dark};
      }
    }
  `}
`