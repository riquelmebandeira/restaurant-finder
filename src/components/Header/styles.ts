import styled from "styled-components";
import img from '../../assets/header-background.png'

export const Header = styled.header`
background-image: url(${img});
background-size: cover;
background-position: center;
height: 37.5rem;
position: relative;
width: 100vw;

.content {
  max-width: 110rem;
  display: flex;
  justify-content: space-between;
  margin: auto;
  padding: 3rem 1rem;
}

.text-box {
  margin-top: -1rem;
  text-align: right;
}

form {
  position: absolute;
  bottom: 6rem;
  left: 50%;
  transform: translateX(-50%);
  width: 100%;
}
`