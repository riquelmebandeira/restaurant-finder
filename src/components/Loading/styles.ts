import styled, { keyframes } from 'styled-components'
import loadingIcon from '../../assets/icons/loading.svg'

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg)
  }
`

export const Loading = styled.img.attrs({
  alt: 'Ã­cone de carregamento',
  src: loadingIcon
})`
  animation: ${rotate} 2s linear infinite;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
