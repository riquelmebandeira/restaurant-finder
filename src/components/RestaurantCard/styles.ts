import styled from 'styled-components'

export const Card = styled.div`
  width: 35rem;
  height: 17rem;
  position: relative;
  border-radius: 0.8rem;

  *:not(h6) {
    width: 100%;
    height: 100%;
  }

  img {
    object-fit: cover;
    border-radius: 0.8rem;
    filter: brightness(60%);
  }

  h6 {
    position: absolute;
    bottom: 1rem;
    left: 1rem;
    color: ${({ theme }) => theme.colors.white};
  }
`
