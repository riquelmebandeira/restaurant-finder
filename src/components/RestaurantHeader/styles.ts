import styled from 'styled-components'

type HeaderProps = {
  backgroundImg: string
}

export const Header = styled.header<HeaderProps>`
  position: relative;
  height: 19rem;
  width: 100%;

  // BACKGROUND IMAGE
  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: url(${(props) => props.backgroundImg});
    background-size: cover;
    background-position: center;
    filter: brightness(50%);
  }
`

export const BackButton = styled.div`
  display: flex;
  cursor: pointer;
  position: absolute;
  top: 5.2rem;
  left: 17.4rem;
  z-index: 100;

  img {
    margin-right: 1.4rem;
  }
`

export const Container = styled.div`
  align-items: center;
  display: flex;
  margin: auto;
  max-width: 55rem;
  padding: 0 1rem;
  position: relative;
  top: 4rem;
  z-index: 100;

  img {
    border-radius: 50%;
    height: 12rem;
    margin-right: 2.5rem;
    width: 12rem;
  }
`
