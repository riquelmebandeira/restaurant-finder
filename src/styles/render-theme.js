import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from './theme'

export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
}

export const renderWithRouterAndTheme = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </ThemeProvider>
  )
}
