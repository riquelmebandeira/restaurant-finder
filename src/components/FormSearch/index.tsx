import { FormEvent, useState } from 'react'
import * as Styled from './styles'

type FormSearchProps = {
  placeholder: string
  icon: string
  alt: string
  setSearchTerm: Function
}

const FormSearch = ({
  placeholder,
  icon,
  alt,
  setSearchTerm
}: FormSearchProps) => {
  const [inputValue, setInputValue] = useState('')

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault()
    setSearchTerm(inputValue)
  }

  return (
    <Styled.Form
      onSubmit={handleSubmit}
      aria-label="form"
    >
      <img
        src={icon}
        alt={alt}
      />
      <input
        type="text"
        placeholder={placeholder}
        value={inputValue}
        onChange={({ target }) => setInputValue(target.value)}
      />
    </Styled.Form>
  )
}

export default FormSearch
