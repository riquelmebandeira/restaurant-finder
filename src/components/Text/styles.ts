import styled, { css } from 'styled-components'
import { theme } from '../../styles/theme'

type TextProps = {
  color?: keyof typeof theme.colors
  fontWeight?: 100 | 400 | 800
  fontSize?: keyof typeof theme.font.sizes
}

export const Text = styled.p<TextProps>`
  ${({ color = 'white', fontWeight = 400, fontSize = 'xsmall' }) => css`
    font-weight: ${fontWeight};
    color: ${theme.colors[color]};
    font-size: ${theme.font.sizes[fontSize]};
  `}
`
