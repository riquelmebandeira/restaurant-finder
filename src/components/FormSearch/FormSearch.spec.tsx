import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { renderTheme } from '../../styles/render-theme'
import FormSearch from '.'

const props = {
  placeholder: 'Encontre um restaurante',
  icon: '',
  alt: 'ícone de pesquisa',
  setSearchTerm: jest.fn()
}

describe('<FormSearch />', () => {
  it('should render correctly', () => {
    renderTheme(<FormSearch {...props} />)

    const icon = screen.getByRole('img', { name: 'ícone de pesquisa' })
    const input = screen.getByPlaceholderText('Encontre um restaurante')

    expect(icon).toBeInTheDocument()
    expect(input).toBeInTheDocument()
  })

  it('should trigger "setSearchTerm" when enter is pressed', () => {
    renderTheme(<FormSearch {...props} />)

    const input = screen.getByPlaceholderText('Encontre um restaurante')

    userEvent.type(input, 'arabek')
    expect(props.setSearchTerm).toBeCalledTimes(0)

    userEvent.type(input, '{enter}')
    expect(props.setSearchTerm).toBeCalledTimes(1)
  })

  it('should match the snapshot', () => {
    const { container } = renderTheme(<FormSearch {...props} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
