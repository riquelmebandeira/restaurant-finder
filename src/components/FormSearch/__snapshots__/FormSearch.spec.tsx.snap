// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSearch /> should match the snapshot 1`] = `
<form
  aria-label="form"
  class="sc-bcXHqe dFNhKi"
>
  <img
    alt="Ã­cone de pesquisa"
    src=""
  />
  <input
    placeholder="Encontre um restaurante"
    type="text"
    value=""
  />
</form>
`;
